.travis/before_install.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -e
+
+case "$TRAVIS_OS_NAME" in
+"linux")
+    sudo apt-get -q update
+    sudo apt-get -q install --no-install-recommends -y curl git gnupg-agent locales pinentry-curses pkg-config rsync software-properties-common
+    sudo apt-get -q clean
+    sudo rm -rf /var/lib/apt/lists/*
```

Comment:
Is this cleanup important and valuable in travis ci? If this is a transient container, then we probably don't need to clean it up, which makes this simpler, and we can avoid unnecessary `sudo`s.

Reply:
Good point, I don't think this is necessary. I took this section directly from our [current `.travis.yml` here](https://github.com/apache/avro/blob/ee4c2aaf4a5a0e1108dc01b157150d4629dd7881/.travis.yml#L35). @Fokko, were you the original author for this (git history seems to suggest so)? Do you think we need to keep these cleanup steps?